exporter/containerimage/writer.go

```diff
@@ -212,7 +212,22 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, refCfg cacheconfig.RefC
 	return out, err
 }
 
-func (ic *ImageWriter) commitDistributionManifest(ctx context.Context, ref cache.ImmutableRef, config []byte, remote *solver.Remote, oci bool, inlineCache []byte, buildInfo []byte) (*ocispecs.Descriptor, *ocispecs.Descriptor, error) {
+func appendExtraLayer(ctx context.Context, ref cache.ImmutableRef, refCfg cacheconfig.RefConfig, descs []ocispecs.Descriptor, sg session.Group) ([]ocispecs.Descriptor, error) {
+	if refCfg.Compression.Type == compression.Nydus {
+		// For nydus format, appending an extra nydus bootstrap layer to the manifest,
+		// this layer represents the whole metadata of filesystem view for the entire image.
+		// The nydus merge operation is fast, so we don't need to create a new cache
+		// record for this extra layer, the layer will be gc after lease expired.
```

Comment:
What stops the the layer from being gc'd before it's used? I didn't see a lease added for it anywhere, is it referenced by something with a lease?

Reply:
Thanks for the reply. I think that the lease was created [here](https://github.com/moby/buildkit/pull/2581/files#diff-9dc5570df38eb8f9d15ac01a23974183e935a6825308f61007b98f6f5278102bL267) to stop the layer from being gc'd before it's pushed. Please help to point out if I have misunderstood something.